src/services/FirebaseService.ts

```diff
@@ -36,15 +37,15 @@ class Firebase {
     return this.auth.createUserWithEmailAndPassword(email, password);
   }
 
-  doSignInWithEmailAndPassword(
+  async doSignInWithEmailAndPassword(
```

Comment:
What is the benefit of making this async? I think we would still need to wait for sign in to succeed before proceeding with anything else right?

Reply:
Ohya, making this async was primarily for the calls `this.auth.setPersistence(...)`. I added `await` because `setPersistance` returns a promise, otherwise, I think we could potentially be returning the call before setPersistence finishes

Reply:
Ah, makes sense